extractor/src/main/scala/com/digitalasset/extractor/json/JsonConverters.scala

```diff
@@ -3,58 +3,68 @@
 
 package com.digitalasset.extractor.json
 
-import com.digitalasset.daml.lf.data.SortedLookupList
+import com.digitalasset.daml.lf.data.{
+  Decimal => LfDecimal,
+  FrontStack,
+  ImmArray,
+  Ref,
+  SortedLookupList,
+  Time
+}
+import com.digitalasset.daml.lf.value.{Value => V}
 import com.digitalasset.extractor.ledger.types.{Identifier, LedgerValue}
 import com.digitalasset.extractor.ledger.types.LedgerValue._
 import com.digitalasset.extractor.writers.postgresql.DataFormatState.MultiTableState
 import io.circe._
 import io.circe.generic.auto._
 import io.circe.generic.semiauto._
 import io.circe.syntax._
+import scalaz.@@
 
 object JsonConverters {
   def toJsonString[A: Encoder](a: A): String = {
     a.asJson.noSpaces
   }
 
-  implicit val recordEncoder: Encoder[Record] = record => {
-    record.fields
-      .foldLeft(JsonObject.empty) {
-        case (acc, field) =>
-          acc.+:(field.label -> field.value.asJson)
-      }
-      .asJson
-  }
+  implicit val recordEncoder: Encoder[OfCid[V.ValueRecord]] = record =>
+    if (record.fields.toSeq.forall(_._1.isDefined))
```

Comment:
we convert `record.fields` to sequence twice for no reason, no matter which if/else branch we go... could reuse the value returned by the 1st `toSeq` call.

I hope we have a test case for a record with one label present `Some(x)`, and one absent `None`. Not sure if `record.fields.toSeq.map(_.asJson).asJson` will do the right thing.

Reply:
> we convert `record.fields` to sequence twice for no reason, no matter which if/else branch we go... could reuse the value returned by the 1st `toSeq` call.

`toSeq` is ridiculously cheap, little value in combining.

> I hope we have a test case for a record with one label present `Some(x)`, and one absent `None`.

Nope! I'll add one.